enterprise/api/v1/views.py

```diff
@@ -0,0 +1,123 @@
+"""
+Views for enterprise api version 1 endpoint.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from rest_framework import filters, permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from django.contrib.auth.models import User
+from django.contrib.sites.models import Site
+
+from enterprise import models
+from enterprise.api.filters import EnterpriseCustomerUserFilterBackend
+from enterprise.api.throttles import ServiceUserThrottle
+from enterprise.api.v1 import serializers
+
+
+class EnterpriseCustomerViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    API views for `enterprise customer` api endpoint.
+    """
+    queryset = models.EnterpriseCustomer.active_customers.all()
+    serializer_class = serializers.EnterpriseCustomerSerializer
+    filter_backends = (filters.OrderingFilter, filters.DjangoFilterBackend)
+    permission_classes = (permissions.IsAuthenticated,)
+    authentication_classes = (OAuth2Authentication, SessionAuthentication,)
+    throttle_classes = (ServiceUserThrottle,)
```

Comment:
@saleem-latif I am seeing the same things in all other views. 
Is it possible to move the common things such as `filter_backends, permission_classes`, `authentication_classes`, `throttle_classes` in separate class that is inherited with `viewsets.ReadOnlyModelViewSet`